### PR TITLE
Set up an error page when the token could not be validated.

### PR DESCRIPTION
**Pull request by Vlatombe**
When using **Token Field Key To Check** and **Token Field Value To Check** to restrict login access to an instance, if a user that doesn't match the specified criteria, you get a HTTP 401 error

Capture d’écran 2024-06-24 à 17 18 23

This introduces a nicer error page, providing a link to log out from the OpenID Connect Provider (if configured).

Capture d’écran 2024-06-25 à 12 52 52

### Testing done

Tested manually with a configured **Token Field Key To Check**/**Token Field Value To Check** and combinations where the user matches the token field or doesn't match it.

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by mikecirioli — APPROVED** [review]

File: src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java [hunk 1]
```diff
@@ -1194,24 +1196,27 @@ public void doLogout(StaplerRequest req, StaplerResponse rsp) throws IOException
 
     @Override
     public String getPostLogOutUrl2(StaplerRequest req, Authentication auth) {
+        Object idToken = req.getAttribute(ID_TOKEN_REQUEST_ATTRIBUTE);
+        Object state = req.getAttribute(STATE_REQUEST_ATTRIBUTE);
+        var openidLogoutEndpoint = maybeOpenIdLogoutEndpoint(
+                Objects.toString(idToken), Objects.toString(state), this.postLogoutRedirectUrl);
+        if (openidLogoutEndpoint != null) return openidLogoutEndpoint;
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java:1203]:
```suggestion
        if (openidLogoutEndpoint != null) {
            return openidLogoutEndpoint;
        }
```

File: src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java [hunk 2]
```diff
@@ -1194,24 +1196,27 @@ public void doLogout(StaplerRequest req, StaplerResponse rsp) throws IOException
...
+        return getFinalLogoutUrl(req, auth);
+    }
+
+    private String maybeOpenIdLogoutEndpoint(String idToken, String state, String postLogoutRedirectUrl) {
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java:1207]:
```suggestion
    @CheckForNull
    private String maybeOpenIdLogoutEndpoint(String idToken, String state, String postLogoutRedirectUrl) {
```

**Review by amuniz — APPROVED** [review]

**Comment by Vlatombe** [comment]
@michael-doubez This is ready to be reviewed

**Review by michael-doubez — APPROVED** [review]
LGTM

**Pull request merged**